test/3_withdrawals.ts

```diff
@@ -454,6 +464,31 @@ describe('Cashier withdrawals ', () => {
           }
         );
 
+        if (i == 0) {
+          // deposits not released before, event should be emitted here
+          let expectedType = 0;
+          eventUtils.assertEventEmitted(
+            txReceipt,
+            VoucherKernel_Factory,
+            eventNames.LOG_FUNDS_RELEASED,
+            (ev) => {
+              assert.isTrue(ev._tokenIdVoucher.eq(voucherID));
+
+              assert.equal(ev._type, expectedType++); // 0 == withdraw, 1 == deposit
+            }
+          );
+        } else {
+          eventUtils.assertEventEmitted(
+            txReceipt,
+            VoucherKernel_Factory,
+            eventNames.LOG_FUNDS_RELEASED,
```

Comment:
I might miss the context, but if Event Type is 1 (deposit), should we use eventNames.LOG_FUNDS_RELEASED (sounds to me like a withdrawal)?

Reply:
`LOG_FUNDS_RELEASED` is ok, since we are actually releasing funds and withdrawing them.

Whenever we are releasing them, those funds can have one of two possible origins:
- funds were sent to the escrow as a payment for the product
- funds were sent to the escrow as a buyer or seller deposit, which are there to prevent misbehaviour

So basicaly `eventType` here just tells what the origin of withdrawn funds is.
But yes, I made a mistake in the comments, since 0 does not represent withdraw but "payment". I will correct that.